src/libraries/System.IO.Pipelines/tests/PipeReaderWriterFacts.cs

```diff
@@ -229,6 +229,34 @@ void ThrowTestException()
             Assert.Single(Regex.Matches(invalidOperationException.StackTrace, "Pipe.GetReadResult"));
         }
 
+        [Fact]
+        public async Task WriteAsync_ThrowsIfReaderCompletedWithException()
+        {
+            void ThrowTestException()
+            {
+                try
+                {
+                    throw new InvalidOperationException("Reader exception");
+                }
+                catch (Exception e)
+                {
+                    _pipe.Reader.Complete(e);
+                }
+            }
+
+            ThrowTestException();
+
+            InvalidOperationException invalidOperationException =
+                await Assert.ThrowsAsync<InvalidOperationException>(async () => await _pipe.Writer.WriteAsync(new byte[1]));
```

Comment:
Do we already have the GetMemory/Advance/FlushAsync version of this test?

Reply:
https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.IO.Pipelines/tests/BackpressureTests.cs#L207